dds/DCPS/transport/rtps_udp/BundlingCacheKey.h

```diff
@@ -27,120 +27,57 @@ namespace DCPS {
-    , addr_guids_(addr_guids)
-#if defined ACE_HAS_CPP11
-    , hash_(calculate_hash())
-#endif
   {
   }
 
-  BundlingCacheKeyCore(const BundlingCacheKeyCore& val)
+  BundlingCacheKey(const BundlingCacheKey& val)
     : src_guid_(val.src_guid_)
     , dst_guid_(val.dst_guid_)
-    , addr_guids_(val.addr_guids_)
-#if defined ACE_HAS_CPP11
-    , hash_(val.hash_)
-#endif
-  {
-  }
-
-  bool operator<(const BundlingCacheKeyCore& rhs) const
-  {
-    int r = std::memcmp(static_cast<const void*>(&src_guid_), static_cast<const void*>(&rhs.src_guid_), 2 * sizeof (GUID_t));
-    if (r < 0) {
-      return true;
-    } else if (r == 0) {
-      return addr_guids_->guids_ < rhs.addr_guids_->guids_;
-    }
-    return false;
-  }
-
-  bool operator==(const BundlingCacheKeyCore& rhs) const
-  {
-    return std::memcmp(static_cast<const void*>(&src_guid_), static_cast<const void*>(&rhs.src_guid_), 2 * sizeof (GUID_t)) == 0 &&
-#if defined ACE_HAS_CPP11
-      addr_guids_->hash() == rhs.addr_guids_->hash() &&
-#endif
-      addr_guids_->guids_ == rhs.addr_guids_->guids_;
-  }
-
-  BundlingCacheKeyCore& operator=(const BundlingCacheKeyCore& rhs)
-  {
-    if (this != &rhs) {
-      const_cast<GUID_t&>(src_guid_) = rhs.src_guid_;
-      const_cast<GUID_t&>(dst_guid_) = rhs.dst_guid_;
-      addr_guids_ = rhs.addr_guids_;
-    }
-    return *this;
-  }
-
-  const GUID_t src_guid_;
-  const GUID_t dst_guid_;
-  RcHandle<ConstSharedRepoIdSet> addr_guids_;
-#if defined ACE_HAS_CPP11
-  const size_t hash_;
-
-  size_t calculate_hash()
   {
-    uint32_t hash = addr_guids_->hash();
-    hash = OpenDDS::DCPS::one_at_a_time_hash(reinterpret_cast<const uint8_t*>(&src_guid_), 2 * sizeof (OpenDDS::DCPS::GUID_t), hash);
-    return static_cast<size_t>(hash);
   }
-#endif
-};
-
-#pragma pack(pop)
 
-struct OpenDDS_Rtps_Udp_Export BundlingCacheKey : public virtual RcObject, public BundlingCacheKeyCore {
-  BundlingCacheKey(const GUID_t& dst_guid, const GUID_t& src_guid, RcHandle<ConstSharedRepoIdSet> addr_guids)
-    : RcObject()
-    , BundlingCacheKeyCore(dst_guid, src_guid, addr_guids)
+  bool operator<(const BundlingCacheKey& rhs) const
   {
-  }
-
-  BundlingCacheKey(const BundlingCacheKey& val)
-    : RcObject()
-    , BundlingCacheKeyCore(val)
-  {
-  }
-
-  inline bool operator<(const BundlingCacheKey& rhs) const
-  {
-    return BundlingCacheKeyCore::operator<(rhs);
+    return std::memcmp(static_cast<const void*>(this), static_cast<const void*>(&rhs), sizeof (BundlingCacheKey)) < 0;
   }
 
   bool operator==(const BundlingCacheKey& rhs) const
   {
-    return BundlingCacheKeyCore::operator==(rhs);
+    return std::memcmp(static_cast<const void*>(this), static_cast<const void*>(&rhs), sizeof (BundlingCacheKey)) == 0;
   }
 
   BundlingCacheKey& operator=(const BundlingCacheKey& rhs)
   {
     if (this != &rhs) {
-      BundlingCacheKeyCore::operator=(rhs);
+      const_cast<GUID_t&>(src_guid_) = rhs.src_guid_;
+      const_cast<GUID_t&>(dst_guid_) = rhs.dst_guid_;
```

Comment:
If the class supports assignment then these aren't really const.  I guess I'd rather see `const` removed from the data members.

Reply:
removing operator= for now